base/compiler/optimize.jl

```diff
@@ -311,6 +311,13 @@ function statement_cost(ex::Expr, line::Int, src::CodeInfo, sptypes::Vector{Any}
                 ftyp = argextype(farg, src, sptypes, slottypes)
             end
         end
+        # Give calls to OpaqueClosures zero cost. The plan is for these to be a single
```

Comment:
It seems like everything below here should be a seperate function.
(Or rather a collection of seperate methods).

Is the reason it isn't to do with compilation costs?

Reply:
No, this could be refactored.